addons/mail/static/src/model/model_manager.js

```diff
@@ -106,81 +108,9 @@ class ModelManager {
-                            // implementation details that read on `this._super` even
-                            // if not set before-hand.
-                            return target[k];
-                        }
-                        return Model.fields[k].get(target);
-                    },
-                    set: (target, k, newVal) => {
-                        if (fieldNames.has(k)) {
-                            throw new Error("Forbidden to write on record field without .update()!!");
-                        } else {
-                            // No crash, we allow these writes due to following concerns:
-                            // - patch() implementation details that write on `this._super`
-                            // - record listeners that need setting on this with `.bind(this)`
-                            target[k] = newVal;
-                        }
-                        return true;
-                    },
-                });
-                record.env = this.env;
-                record.localId = Model._createRecordLocalId(data);
-                if (Model.get(record.localId)) {
-                    throw Error(`A record already exists for model "${Model.modelName}" with localId "${record.localId}".`);
-                }
-                // Contains field values of record.
-                record.__values = {};
-                // Contains revNumber of record for checking record update in useStore.
-                record.__state = 0;
-
-                Model.__records[record.localId] = record;
-                record.init();
-
-                // Ensure X2many relations are Set initially (other fields can stay undefined).
-                for (const field of fields) {
-                    if (field.fieldType === 'relation') {
-                        if (['one2many', 'many2many'].includes(field.relationType)) {
-                            record.__values[field.fieldName] = new Set();
-                        }
-                    }
-                }
-
-                const data2 = {};
-                for (const field of fields) {
-                    if (field.fieldName in data) {
-                        data2[field.fieldName] = data[field.fieldName];
-                    } else {
-                        data2[field.fieldName] = field.default;
-                    }
-                }
-
-                for (const field of fields) {
-                    if (field.compute || field.related) {
-                        // new record should always invoke computed fields.
-                        this._registerToComputeField(record, field);
-                    }
-                }
-
-                this.update(record, data2);
-
-                records.push(record);
-            }
-            return isMulti ? records : records[0];
-        });
+        const res = this._create(Model, data);
+        this._flushUpdateCycle();
+        return res;
```

Comment:
I prefer this:
```js
return this._updateCycle(() => this._create(Model, data));
```

Because it's clear that this function makes an update cycle, and "flush" is a detail of the update cycle itself, so it would make more sense to is it once in the `_updateCycle` method.

Reply:
Ok for other comments, will do progressively.

Just this one might be critical for future improvements.
I think we should give control to update cycle to business code eventually, instead of tying it up to some model methods only.

This would allow to more easily expand it, for example it would make sense that update cycle is only flushed once, at the end of a handler (once after init, and the once after each (notification, resize, ...) handler or once after each user action handler).
I think this might instantly allow async update cycle as well, the flush can simply be delayed.

This is inspired by python that also does `flush` the least possible (once at the end of the request ideally) but it can be called manually if need be for specific reasons.

Reply:
I don't like explicit `_flushUpdateCycle` because it spreads too much the dynamic aspect of the update cycle everywhere, especially so close to business code.

The business code should not be concerned by update cycle, it's the model manager that just handles it. Optimizations such as the ones you named should be solved in other ways, like batching some method calls together in an update cycle (instead of manually tell when to flush or not).

Reply:
This change is the goal of the PR. The previous update cycle was incorrectly too rigid.
Having a cycle made it look like we needed a clear "start" and "end" of cycle, which was somewhat needed for technical reasons before, but it is not the case anymore.

The only thing that matters is being able to delay compute for performance reasons, and the only action that makes sense in this context is to be able to get the computes done when necessary, which is the goal of `flush`.